src/components/timeline/TimelineHistogram.svelte

```diff
@@ -0,0 +1,419 @@
+      }
+    });
+
+    activityHistMax = Math.max(...activityHistValues);
+
+    // Compute constraint violations histogram
+    constraintViolationsToRender = [];
+    constraintViolations.forEach(violation => {
+      violation.windows.forEach(w => {
+        const xStart = xScaleMax(w.start);
+        const xEnd = xScaleMax(w.end);
+        const clampedStart = xStart < 0 ? 0 : xStart;
+        const clampedEnd = xEnd > drawWidth ? drawWidth : xEnd;
+        const width = clampedEnd - clampedStart;
+        const clampedWidth = width <= 0 ? 5 : width;
+        constraintViolationsToRender.push({ width: clampedWidth, x: clampedStart });
+      });
+    });
+  }
+
+  function brushed({ selection }) {
+    if (!selection || selection[1] - selection[0] === 0) {
+      brush.attr('display', 'none');
+    } else {
+      brush.attr('display', 'initial');
+    }
+
+    brush.selectAll('.selection').attr('rx', '4px');
+
+    brush
+      .selectAll('.handle')
+      .attr('height', `${selectorHandleHeight}px`)
+      .attr('y', `${drawHeight / 1.9 / 2}px`)
+      .attr('rx', '4px');
+  }
+
+  function onMouseMove(x: number, y: number, checkY: boolean = true) {
+    const histRect = histogramContainer.getBoundingClientRect();
+    const mouseWithinLeftHorizontalHistogramBounds = x >= histRect.x;
+    const mouseWithinRightHorizontalHistogramBounds = x <= histRect.right;
+    const mouseWithinHorizontalHistogramBounds =
+      mouseWithinLeftHorizontalHistogramBounds && mouseWithinRightHorizontalHistogramBounds;
+    const mouseWithinVerticalHistogramBounds = checkY ? y >= histRect.y && y <= histRect.bottom : true;
+
+    // Check if mouse is within histogram position bounds
+    if (mouseWithinVerticalHistogramBounds && mouseWithinHorizontalHistogramBounds) {
+      // Update hover cursor
+      timelineHovering = true;
+      cursorLeft = x - histRect.left;
+      const cursorTime = xScaleMax.invert(cursorLeft);
+      cursorTooltip = getDoyTime(cursorTime, false);
+
+      // Only dispatch a cursor change if we're just hovering
+      if (!movingSlider && !drawingRange && !resizingSliderLeft && !resizingSliderRight) {
+        dispatch('cursorTimeChange', cursorTime);
+      }
+    } else {
+      timelineHovering = false;
+      dispatch('cursorTimeChange', null);
+    }
+  }
+
+  function onWindowMouseMove(e: MouseEvent) {
+    onMouseMove(e.x, e.y);
+  }
+</script>
+
+<div
+  bind:this={histogramContainer}
+  class="timeline-histogram"
+  style={`margin-left: ${marginLeft}px; width: ${drawWidth}px; height: ${drawHeight}px;`}
+>
+  <div class="timeline-histogram-background" />
+  <div
+    use:tooltip={{
+      content: cursorTooltip,
+      hideOnClick: false,
+      placement: 'top',
+      showOnInit: true,
+      triggerTarget: histogramContainer,
+    }}
+    class="timeline-histogram-cursor"
+    style={`left: ${cursorLeft}px; opacity: ${brushing ? 0 : 1}`}
+    hidden={!cursorVisible && !timelineHovering}
```

Comment:
I'm wondering if we should still show the tooltip if the timeline cursor is not visible? The line still shows in the time range slider even if the timeline cursor is hidden. Or maybe we should just hide the line all together if the cursor is hidden.

Reply:
hmm no not that. Do you think it makes sense to still show the line in the slider when the cursor is hidden?

Reply:
Oh like if you hide the main timeline cursor via keyboard shortcut (I don't remember what that shortcut is)?

Reply:
Ah `ctrl-t` – yeah should at least hide the mirrored cursor in the slider when hovering over the timeline when main cursor is disabled. But should we hide that cursor for both timeline and slider when cursor is off?